src/coreclr/tools/Common/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -200,7 +200,26 @@ private EcmaModule AddModule(string filePath, string expectedSimpleName, bool us
                     throw new NotSupportedException($"Error: C++/CLI is not supported: '{filePath}'");
 #endif
 
-                    pdbReader = PortablePdbSymbolReader.TryOpenEmbedded(peReader, GetMetadataStringDecoder()) ?? OpenAssociatedSymbolFile(filePath, peReader);
+                    try
+                    {
+                        pdbReader = PortablePdbSymbolReader.TryOpenEmbedded(peReader, GetMetadataStringDecoder())
+                                    ?? OpenAssociatedSymbolFile(filePath, peReader);
+                    }
+                    catch (BadImageFormatException e)
+                    {
+                        if (e.Message.Equals("Invalid directory size."))
+                        {
+                            int actualDbgDirSize = peReader.PEHeaders.PEHeader.DebugTableDirectory.Size;
+
+                            // This comes from the Size property of the DebugDirectoryEntry class.
+                            int expectedDbgDirSizeBase = 28;
+
+                            Console.WriteLine("WARNING: This image's Debug Directory Entry size should be a"
+                                            + $" multiple of {expectedDbgDirSizeBase}, but was {actualDbgDirSize}."
+                                            + " Debugging tools may not work correctly with this image.");
```

Comment:
```suggestion
                                            + " Diagnostic tools may not work correctly with this image.");
```
It is also profilers, etc.